src/core/mesh/qgstriangularmesh.cpp

```diff
@@ -223,6 +251,31 @@ QList<int> QgsTriangularMesh::faceIndexesForRectangle( const QgsRectangle &recta
   return mSpatialIndex.intersects( rectangle );
 }
 
+QVector<QVector3D> QgsTriangularMesh::vertexNormales( float vertScale ) const
+{
+  QVector<QVector3D> normales( vertices().count(), QVector3D( 0, 0, 0 ) );
+
+  for ( const auto &face : triangles() )
+  {
+    for ( int i = 0; i < 3; i++ )
+    {
+      int index1( face.at( i ) );
+      int index2( face.at( ( i + 1 ) % 3 ) );
+      int index3( face.at( ( i + 2 ) % 3 ) );
```

Comment:
why % 3 ?

Reply:
if i = 1 or  to access  to the index 0 with i+2
if i = 2 to access to the index 0 and 1 with i+1 and i+2

Reply:
of course - sorry I have misread the code...